src/lib/parser/statement/mod.rs

```diff
@@ -4,15 +4,17 @@ mod parse;
 mod splitter;
 
 pub(crate) use self::{
-    parse::parse, splitter::{StatementError, StatementSplitter},
+    parse::parse, splitter::{StatementVariant, StatementError, StatementSplitter},
 };
 use shell::flow_control::Statement;
 
 /// Parses a given statement string and return's the corresponding mapped
 /// `Statement`
-pub(crate) fn parse_and_validate<'a>(statement: Result<String, StatementError>) -> Statement {
+pub(crate) fn parse_and_validate<'a>(statement: Result<StatementVariant, StatementError>) -> Statement {
     match statement {
-        Ok(statement) => parse(statement.as_str()),
+        Ok(StatementVariant::And(statement)) => parse(statement),
+        Ok(StatementVariant::Or(statement)) => parse(statement),
```

Comment:
You'll need to create a Statement::And/Or directly, rather than passing it off to be parsed into a Statement.

Reply:
How do I do that, since `Statement::{And,Or}` both take in a `Box<Statement>`?

Reply:
```rust
Box::new(Statement::And(parse(statement)))
```

Reply:
Thank you